### ISSUE TITLE
[BUG] Join Index Name Regression

### ISSUE THREAD

**Issue by quasiben** [rapidsai/cudf#1945]
When joining dataframes the index name is no longer None but instead a random number.  

In cudf 0.7.X the resulting index was None. 

```python
In [10]: left_nrows, right_nrows, left_nkeys, right_nkeys = (5, 5, 4, 4)

In [11]: left = cudf.DataFrame(
    ...:         {
    ...:             "x": np.random.randint(0, left_nkeys, size=left_nrows),
    ...:             "a": np.arange(left_nrows),
    ...:         }.items()
    ...:     )

In [12]: right = cudf.DataFrame(
    ...:         {
    ...:             "x": np.random.randint(0, left_nkeys, size=left_nrows),
    ...:             "a": np.arange(left_nrows),
    ...:         }.items()
    ...:     )

In [13]: left.set_index("x").join(right.set_index("x"), how="inner", sort=True, lsuffix="l", rsuffix="r").to_pandas()
Out[13]:
   al  ar
0   2   3
1   3   0
1   3   1
2   4   4
3   0   2
3   1   2

In [14]: cudf.__version__
Out[14]: '0.7.1+0.g7ee0348.dirty'
```

in branch-0.8 the index is a random number:

```
In [1]: import cudf

In [2]: cudf.__version__
Out[2]: '0.7.0+1127.g7312d4a'

In [3]: import numpy as np

In [5]: left_nrows, right_nrows, left_nkeys, right_nkeys = (5, 5, 4, 4)

In [6]: left = cudf.DataFrame(
   ...:         {
   ...:             "x": np.random.randint(0, left_nkeys, size=left_nrows),
   ...:             "a": np.arange(left_nrows),
   ...:         }.items()
   ...:     )

In [7]: right = cudf.DataFrame(
   ...:         {
   ...:             "x": np.random.randint(0, left_nkeys, size=left_nrows),
   ...:             "a": np.arange(left_nrows),
   ...:         }.items()
   ...:     )

In [8]: left.set_index("x").join(right.set_index("x"), how="inner", sort=True, lsuffix="l", rsuffix="r").to_pandas()
Out[8]:
            al  ar
1875641867
1            0   4
1            1   4
1            3   4
1            4   4
3            2   3
```

**Comment by thomcom** [comment]
There is a good chance that this is related to https://github.com/rapidsai/cudf/pull/1850

**Comment by kkraus14** [comment]
Fixed by #1982
